packages/dds/tree/src/test/snapshots/summary.spec.ts

```diff
@@ -3,27 +3,47 @@
  * Licensed under the MIT License.
  */
 
+import { TreeCompressionStrategy } from "../../feature-libraries/index.js";
+import { SharedTreeFormatVersion, type SharedTreeOptions } from "../../shared-tree/index.js";
 import { useSnapshotDirectory } from "./snapshotTools.js";
 import { generateTestTrees } from "./testTrees.js";
 import { takeSummarySnapshot } from "./utils.js";
 
 describe("snapshot tests", () => {
-	useSnapshotDirectory();
-	for (const useUncompressedEncode of [false, true]) {
-		const compressionType = useUncompressedEncode ? "uncompressed" : "default compression";
-		describe(`${compressionType}`, () => {
-			const testTrees = generateTestTrees(useUncompressedEncode);
+	for (const treeEncodeType of [
+		TreeCompressionStrategy.Compressed,
+		TreeCompressionStrategy.Uncompressed,
+	]) {
+		// Friendly description of tree encoding type
+		const treeEncodeKey = TreeCompressionStrategy[treeEncodeType];
+		for (const formatVersionKey of Object.keys(SharedTreeFormatVersion)) {
```

Comment:
I wish there were a compile-type variant of Object.keys that returned an array (or tuple!) where the elements have the more precise type. Can't do that with a language that elides all typing info at runtime though!